brainiak/reprsimil/brsa.py

```diff
@@ -767,23 +1203,24 @@ def _fit_RSA_UV(self, X, Y, X0,
                 XTY, XTDY, XTFY, X0TX0, X0TDX0, X0TFX0,
                 XTX0, XTDX0, XTFX0, X0TY, X0TDY, X0TFY,
                 X, Y, X0, idx_param_sing,
-                l_idx, n_C, n_T, n_V, n_l, n_run, n_base, rank)
+                l_idx, n_C, n_T, n_V, n_l, n_run, n_X0, rank)
 
         current_logSNR2 = -current_logSigma2
         norm_factor = np.mean(current_logSNR2)
         current_logSNR2 = current_logSNR2 - norm_factor
+        X_res = None
 
         # Step 2 fitting, which only happens if
         # GP prior is requested
         if GP_space:
-            current_vec_U_chlsk_l, current_a1, current_logSNR2, X0 \
+            current_vec_U_chlsk_l, current_a1, current_logSNR2, X_res\
                 = self._fit_diagV_noGP(
                     XTY, XTDY, XTFY, YTY_diag, YTDY_diag, YTFY_diag,
-                    XTX, XTDX, XTFX, X, Y, X0, D, F, run_TRs,
+                    XTX, XTDX, XTFX, X, Y, X_base, X_res, D, F, run_TRs,
                     current_vec_U_chlsk_l,
                     current_a1, current_logSNR2,
                     idx_param_fitU, idx_param_fitV,
-                    l_idx, n_C, n_T, n_V, n_l, n_run, n_base, rank)
+                    l_idx, n_C, n_T, n_V, n_l, n_run, n_X0, rank)
 
             current_GP[0] = np.log(np.min(
```

Comment:
The length does vary depending on the GP options. But the values always mean the same thing: the 0-th is always the length scale in voxel distance, the 1-st is always the length scale in intensity difference. I think I did this just to save the possible if/else clauses. Otherwise, I need to write separate lines for whether GP_inten is set to True in many places.